### ISSUE TITLE
Suggestions for collaborating

### ISSUE THREAD

**Issue by remembertoremind** [smallhadroncollider/taskell#71]
Hi! I know you are busy with work, but I just wanted to write this before I forgot it. Besides, I _really_ like taskell, so if these features could be implemented there would be a chance for my colleagues to actually use it. So, here are my suggestions. Forgive me if these suggestions dont make much sense. I'm not a dev myself.

1. **Lock the taskell.md file when it's open** in taskell. Currently it's possible to open the taskell.md from different intances and there are no limitations for two different instances to edit the same file
2. **Assign tasks and ability to print these to the terminal**. Like, if several tasks had different names associated and it does not make sense to look through all tasks to find out where you have been tagged. 

Sample:
```
## To Do

- Find treasure
    @ 2020-03-29
    * [ ] @Christian: Find map
    * [ ] @John: Talk to pirates
- Daily todo
    @ 2020-03-30
    * [x] >Christian: lol
    * [ ] >Christian: Find the code
    * [x] >Christian: dont mess with the code
    * [ ] >John: Mess with code
    * [ ] >John: Blame Christian

## Done
```

`taskell --input taskell.md --section todo --completed --user Christian` output something like the example below. 

```
## To Do

- Daily todo
    @ 2020-03-30
    * [x] >Christian: lol
    * [x] >Christian: dont mess with the code

```
`taskell --input taskell.md --section todo --pending --user John` output something like this :

```
- Find treasure
    @ 2020-03-29
    * [ ] @John: Talk to pirates
- Daily todo
    @ 2020-03-30
    * [ ] >John: Mess with code
    * [ ] >John: Blame Christian
```

I'm pretty sure this could be done with `awk`, but I think others would also like the option :)

**Comment by smallhadroncollider** [comment]
I like the idea of being able to assign users to different tasks, but certainly needs some consideration before starting on it.

e.g. What if you had two John's? Use emails instead? Or leave it up to user to decide - might get tricky if multiple people are editing it.

**Comment by remembertoremind** [comment]
Yeah, I think it would help to ensure all tasks are being dealt with.

I was thinking you could have a global list or a list pr project in your git repo with emails and chosen username so two Johns would be different in some way. The most important thing  is to be able to cat out the project to see if you're mentioned, so you could miss a kanban meeting and still know what's up and deal with the info in some way like

```
#!/bin/bash
randomwall=$(ls -1 $HOME/Pictures/Wallpaper/ | shuf -n 1)
taskellpending=$(taskell --input taskell.md --section todo --pending --user John@john.john)
    convert -geometry "x1080"\
                -fill '#00000099'\
                -draw "rectangle 50,50 530,1030"\
                -pointsize 20 \
                -fill '#ffff00'\
                -font 'Terminus-(TTF)-Bold' \
                -pointsize '25' \
                -draw "text 70,100 '$taskellpending'" \
        $HOME/Pictures/Wallpaper/$randomwall $HOME/Pictures/wallpaper
        feh --bg-scale $HOME/Pictures/wallpaper
```

**Comment by smallhadroncollider** [comment]
I was thinking about this the other day. I think the markdown format could be:

```markdown
# Title (not yet supported)

## Collaborators

- @bob (any other information)
- @reni (full name, email, etc.)

---

## First List

- Do a thing @bob
- Do another thing @reni @bob
```

Then you could easily define collaborators as part of the file.

**Comment by remembertoremind** [comment]
Hm.. Well, you could define contacts somewhere else in your taskell.md but I think it would make more sense to have _one_ file with contacts. Either in the project directory or a global. Perhaps defined in the top of taskell.md - like metadata?


taskell.md

`%
% collaborators = $HOME/Projects/taskell/coworkers.md
%
# Title (not yet supported)

## Collaborators

- @bob (any other information)
- @reni (full name, email, etc.)

---

## First List

- Do a thing @bob
- Do another thing @reni @bob`

coworkers.md

`bob = Bob Oderkerk Simpleton boobs@firm.org
reni = Reni Wong Here no@firm.org`
